specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp

```diff
@@ -204,17 +168,46 @@ model ChatCompletionsOptions {
   @projectedName("json", "dataSources")
   @projectedName("csharp", "InternalAzureExtensionsDataSources")
   dataSources?: AzureChatExtensionConfiguration[];
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("If provided, the configuration options for available Azure OpenAI chat enhancements.")
+  enhancements?: AzureChatEnhancementConfiguration;
```

Comment:
I don't want to add to the pile here, but if we are incurring into already plenty of API breaking changes we may want to go all in and describe `AzureChatCompletionsOptions` in a way that composes the fields closer to what is done in the swagger:

```ts
model ChatCompletionsOptionsBase {
  // the usual fields but we move `dataSources` to an azure dedicated model for better compo
  // We also take all the fields that will show up in the model below
  // ...
}

model ChatCompletionsOptions {
  ...ChatComplettionsOptionsBase;
  messages: ChatMessages[];
  n: integer;
  seed: integer;
  response_format: ChatCompletionsResponseFormat;
  tools: ChatCompletionsToolDefinition[];
  tool_choice: ChatCompletionsToolSelection;
  functions: FunctionDefinition[];
  function_call: FunctionCallConfig;
}

model AzureChatCompletionsOptions {
  ...ChatCompletionsOptionsBase;
  messages: ChatMessages[]; 
  dataSources?: AzureChatExtensionConfiguration[];
  enhancements?: AzureChatExtensionEnhancements;
}
```